File: app/src/main/java/example/com/crackle/database/GenreConverter.java
package example.com.crackle.database;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;

public class GenreConverter {

    @TypeConverter
    public static ArrayList<Integer> fromInt(String value) {
        Type listType =new TypeToken<ArrayList<Integer>>(){}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromArrayList(ArrayList<Integer> list) {
        return new Gson().toJson(list);
    }
}

File: app/src/main/java/example/com/crackle/database/GenreConverter.kt
package example.com.crackle.database

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.util.*

class GenreConverter {

    @TypeConverter
    fun fromInt(value: String): ArrayList<Int>? {
        val listType = object : TypeToken<ArrayList<Int>>() {

        }.type
        return Gson().fromJson<ArrayList<Int>>(value, listType)
    }

    @TypeConverter
    fun fromArrayList(list: ArrayList<Int>): String {
        return Gson().toJson(list)
    }
}
